File: mod/SharedStatus.cs
using System;
using UnityEngine;
using Verse;
namespace Concord {
 [Serializable] public class LinkStatus {public string source="shared-link-telemetry",epoch,food,rest;public int tick;}
 [Serializable] public class SharedStatus {public string pawn,name,source,epoch,food,rest;public int tick;public bool fresh;}
 public static class LinkTelemetry {
  // Deliberate game design: linked crew share only these coarse bands.
  private static string Band(RimWorld.Need need){return need==null?"unknown":need.CurLevelPercentage<0.2f?"urgent":need.CurLevelPercentage<0.5f?"low":"satisfied";}
  public static string Json(Pawn p,string epoch){return JsonUtility.ToJson(new LinkStatus {epoch=epoch,tick=Find.TickManager.TicksGame,food=Band(p.needs.food),rest=Band(p.needs.rest)});}
 }
}

File: src/shared-status.ts
import type {Domain,GameState} from './protocol.js';
export type NeedBand='satisfied'|'low'|'urgent'|'unknown';
export type LinkStatus={source:'shared-link-telemetry';epoch:string;tick:number;food:NeedBand;rest:NeedBand};
export type SharedStatus=LinkStatus&{pawn:string;name:string;fresh:boolean};
const band=(x:unknown):NeedBand=>['satisfied','low','urgent'].includes(String(x))?x as NeedBand:'unknown';
/** Explicitly shared telemetry, never inferred from private facts or appearance.
 * Old epochs, future readings and observations older than 120 ticks are unknown. */
export function sharedStatus(d:Domain,g:GameState):SharedStatus[]{
 return Object.values(d.characters).map(c=>{
  const s=g.pawns.find(p=>p.id===c.id)?.linkStatus;
  const fresh=!!s&&s.source==='shared-link-telemetry'&&s.epoch===g.epoch&&Number.isInteger(s.tick)&&s.tick>=0&&s.tick<=g.ticks&&g.ticks-s.tick<=120;
  return {pawn:c.id,name:c.name,source:'shared-link-telemetry',epoch:g.epoch,tick:fresh?s!.tick:g.ticks,fresh,
   food:fresh?band(s!.food):'unknown',rest:fresh?band(s!.rest):'unknown'};
 });
}
